### PR TITLE
Improve build workflow a bit

### PR DESCRIPTION
**Pull request by szepeviktor**
### Description

- upgrade actions
- homogeneous YAML lists
- use GHA working directory
- use the new `github.token`

### What is the purpose of this pull request? 

- [ ] New feature
- [ ] Documentation update
- [ ] Bug fix
- [ ] Refactor
- [ ] Release
- [X] Other: chore

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,32 +27,37 @@ jobs:
...
+        working-directory: app
+        run: npm install
 
       - name: Set environment variables
+        working-directory: app
         run: |
-          echo "VITE_SUPABASE_API_URL=${{ secrets.SUPABASE_API_URL }}" >> app/.env
-          echo "VITE_SUPABASE_ANON_KEY=${{ secrets.SUPABASE_ANON_KEY }}" >> app/.env
-          echo "VITE_MIXPANEL_TOKEN=${{ secrets.MIXPANEL_TOKEN }}" >> app/.env
-          cat app/.env
```
Comment (szepeviktor) [inline .github/workflows/build.yml:53]: This was not a good idea.

Comment (Kitenite) [reply .github/workflows/build.yml:53]: Oof good catch, must've been left over from testing. Will rotate keys

Comment (szepeviktor) [reply .github/workflows/build.yml:53]: Keys are not displayed because GHA never displays secrets.

Comment (szepeviktor) [reply .github/workflows/build.yml:53]: image

Comment (Kitenite) [reply .github/workflows/build.yml:53]: That's smart of them. Even better!

Comment (szepeviktor) [reply .github/workflows/build.yml:53]: Try `tac` 🙃

**Review by Kitenite — APPROVED** [review]
Test build works here: https://github.com/onlook-dev/onlook/actions/runs/10647833173/job/29516209344

Thank you for the improvement!

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**